### DISCUSSION TITLE
Doxygen-style and VSCode compatible API documentation

### DISCUSSION THREAD

**Discussion by frankonski** [HomeSpan/HomeSpan#792] [Ideas]
Hi,

Is there any interest in having HomeSpan's header files and inline comments upgraded a bit, using Doxygen-style comments, so that when using VSCode, one can have online help by simple hover over the API?

**Without proper comments**
![image](https://github.com/HomeSpan/HomeSpan/assets/26729481/db38401d-7642-4b4a-b1ca-f121ae44a1ef)

**With Doxygen-style comments**
![image](https://github.com/HomeSpan/HomeSpan/assets/26729481/05a451db-e52a-468b-978d-f89abbd2746b)

If yes, PRs to come.

**Comment by HomeSpan** [comment]
No, I've played around with that before and find it clashes too much with how I like to write comments and code.  I appreciate the utility, but not really something I'm keen on adding (at least at this point).

**Comment by frankonski** [comment]
Understood. Thanks anyway!
